cnf-certification-test/accesscontrol/securitycontextcontainer/securitycontextcontainer.go

```diff
@@ -0,0 +1,454 @@
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.SELinuxOptions != nil {
+		containerSCC.SeLinuxContextPresent = OK
+	}
+	containerSCC.AllVolumeAllowed, containerSCC.HostDirVolumePluginPresent = AllVolumeAllowed(pod.Spec.Volumes)
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.RunAsUser != nil {
+		containerSCC.RunAsUserPresent = OK
+	} else {
+		containerSCC.RunAsUserPresent = NOK
+	}
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.FSGroup != nil {
+		containerSCC.FsGroupPresent = OK
+	} else {
+		containerSCC.FsGroupPresent = NOK
+	}
+	return CheckCategory(pod.Spec.Containers, containerSCC, pod.Name, pod.Namespace)
+}
+
+//nolint:gocyclo,funlen
+func compareCategory(refCategory, containerSCC *ContainerSCC, id CategoryID) bool {
+	result := true
+	tnf.ClaimFilePrintf("Testing if pod belongs to category %s", &id)
+	if refCategory.AllVolumeAllowed == containerSCC.AllVolumeAllowed {
+		tnf.ClaimFilePrintf("AllVolumeAllowed = %s - OK", containerSCC.AllVolumeAllowed)
+	} else {
+		result = false
+		tnf.ClaimFilePrintf("AllVolumeAllowed = %s but expected >=<=%s -  NOK", containerSCC.AllVolumeAllowed, refCategory.AllVolumeAllowed)
+	}
+	if refCategory.RunAsUserPresent == containerSCC.RunAsUserPresent {
+		tnf.ClaimFilePrintf("RunAsUserPresent = %s - OK", containerSCC.RunAsUserPresent)
+	} else {
+		tnf.ClaimFilePrintf("RunAsUserPresent = %s but expected  %s - NOK", containerSCC.RunAsUserPresent, refCategory.RunAsUserPresent)
+		result = false
+	}
+	if refCategory.RunAsNonRoot >= containerSCC.RunAsNonRoot {
+		tnf.ClaimFilePrintf("RunAsNonRoot = %s - OK", containerSCC.RunAsNonRoot)
+	} else {
+		tnf.ClaimFilePrintf("RunAsNonRoot = %s but expected  %s - NOK", containerSCC.RunAsNonRoot, refCategory.RunAsNonRoot)
+		result = false
+	}
+	if refCategory.FsGroupPresent == containerSCC.FsGroupPresent {
+		tnf.ClaimFilePrintf("FsGroupPresent  = %s - OK", containerSCC.FsGroupPresent)
+	} else {
+		tnf.ClaimFilePrintf("FsGroupPresent  = %s but expected  %s - NOK", containerSCC.FsGroupPresent, refCategory.FsGroupPresent)
+		result = false
+	}
+	if refCategory.HaveDropCapabilities == containerSCC.HaveDropCapabilities {
+		tnf.ClaimFilePrintf("DropCapabilities list - OK")
+	} else {
+		tnf.ClaimFilePrintf("HaveDropCapabilities = %s but expected  %s - NOK", containerSCC.HaveDropCapabilities, refCategory.HaveDropCapabilities)
+		tnf.ClaimFilePrintf("its didnt have all the required (MKNOD, SETUID, SETGID, KILL)/(ALL) drop value ")
+		result = false
+	}
+	if refCategory.HostDirVolumePluginPresent == containerSCC.HostDirVolumePluginPresent {
+		tnf.ClaimFilePrintf("HostDirVolumePluginPresent = %s - OK", containerSCC.HostDirVolumePluginPresent)
+	} else {
+		tnf.ClaimFilePrintf("HostDirVolumePluginPresent = %s but expected  %s - NOK", containerSCC.HostDirVolumePluginPresent, refCategory.HostDirVolumePluginPresent)
+		result = false
+	}
+	if refCategory.HostIPC >= containerSCC.HostIPC {
```

Comment:
Add comment:
// HostIPC  is true if the HostIPC field is set to true, false otherwise.